src/tests/Directory.Build.targets

```diff
@@ -22,6 +22,20 @@
     <NoWarn Condition="'$(IsMergedTestRunnerAssembly)' == 'true'">$(NoWarn);CS2008</NoWarn>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <IlcCompileDependsOn Condition="'$(CustomIlcCompileDependsOn)' == 'true'">Compile;ComputeIlcCompileInputs;SetupOSSpecificProps;PrepareForILLink</IlcCompileDependsOn>
+  </PropertyGroup>
+
+  <Target Name="_SetCustomIlcPath" Condition="'$(CustomIlcCompileDependsOn)' == 'true'" BeforeTargets="SetupProperties">
+    <PropertyGroup>
+      <IlcPath>$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'coreclr', '$(HostOS).$(BuildArchitecture).$(CoreCLRConfiguration)', 'ilc'))</IlcPath>
+      <IlcToolsPath>$(IlcPath)</IlcToolsPath>
+      <IlcSdkPath>$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'coreclr', '$(TargetOS).$(TargetArchitecture).$(CoreCLRConfiguration)', 'aotsdk'))</IlcSdkPath>
```

Comment:
Is this the correct path? I assume the eventual goal is not to compile ILCompiler for iOS and so this should rather point to the runtime pack artifacts.

Reply:
Good point, let's see. The property `$(HostOS)-$(BuildArchitecture)` is set to `osx-x64`, while the property `$(TargetOS)-$(TargetArchitecture)` is set to `ios-arm64` for ios build. The `IlcPath` always points to the host platform. My understanding is that the `IlcSdkPath` should point to the runtime pack built for the target platform, in this case `ios-arm64`.

Reply:
> My understanding is that the IlcSdkPath should point to the runtime pack built for the target platform, in this case ios-arm64.

Correct. However, this points to the iOS ILCompiler package which is something that we don't really want to build, use, or support. The equivalent of `aotsdk` part should be built with the `clr.nativeaotruntime` subset (IIRC) and it should exist somewhere else in the artifacts (where it gets collected when building pack with `BuildNativeAOTRuntimePack=true`).

Reply:
Thank you for the clarification, we shouldn't use the iOS ILC package generally. The `CoreCLRAotSdkDir` property could be used instead, which points to `artifacts/bin/coreclr/osx.x64.Release/aotsdk/` in this case.

Reply:
If it's the same place where the `pack` target picks it then we can close this comment thread :-)

Reply:
Probably not a good path as I got `building for iOS, but linking in object file built for macOS`.

Reply:
Well, perhaps `artifacts/bin/coreclr/ios.arm64.Release/aotsdk/` is the correct path to start with. I just was not sure whether it's the same one that's used for building the runtime packs or not. (It should be, to ensure that we test the same bits.)

Reply:
The used path should match the runtime packs. Currently, we build the ILC for the host and for the target, and it will be improved.